linux-tkg-config/prepare

```diff
@@ -6,11 +6,119 @@ ver58=18
-  if [ -e "$_EXT_CONFIG_PATH" ]; then
-    source "$_EXT_CONFIG_PATH" && msg2 "External configuration file $_EXT_CONFIG_PATH will be used to override customization.cfg values.\n"
+_distro_prompt() {
+  
+  while true; do
+    echo "Which linux distribution are you running ?"
+    echo "if it's not on the list, chose the closest one to it: Fedora/Suse for RPM, Ubuntu/Debian for DEB"
+    echo "   1) Debian"
+    echo "   2) Fedora"
+    echo "   3) Suse"
+    echo "   4) Ubuntu"
+    read -p "[1-4]: " _distro_index
+
+    if [ "$_distro_index" = "1" ]; then
+      _distro="Debian"
+      break
+    elif [ "$_distro_index" = "2" ]; then
+      _distro="Fedora"
+      break
+    elif [ "$_distro_index" = "3" ]; then
+      _distro="Suse"
+      break
+    elif [ "$_distro_index" = "4" ]; then
+      _distro="Ubuntu"
+      break
+    else
+      echo "Wrong index."
+    fi
+  done
+  
+}
+
+_install_dependencies() {
+  if [ "$_compiler_name" = "llvm" ]; then
+    clang_deps="llvm clang lld"
+  fi
+  if [ "$_distro" = "Ubuntu" ]; then
+    msg2 "Installing dependencies"
+    sudo apt install git build-essential fakeroot libncurses5-dev libssl-dev ccache bison flex qtbase5-dev ${clang_deps} -y
+  elif [ "$_distro" = "Debian" ]; then
+    msg2 "Installing dependencies"
+    sudo apt install git wget build-essential fakeroot libncurses5-dev libssl-dev ccache bison flex qtbase5-dev bc rsync kmod cpio libelf-dev ${clang_deps} -y
+  elif [ "$_distro" = "Fedora" ]; then
+    msg2 "Installing dependencies"
+    if [ $(rpm -E %fedora) = "32" ]; then
+      sudo dnf install fedpkg fedora-packager rpmdevtools ncurses-devel pesign grubby qt5-devel libXi-devel gcc-c++ git ccache flex bison elfutils-libelf-devel openssl-devel dwarves rpm-build ${clang_deps} -y
+    else
+      sudo dnf install fedpkg fedora-packager rpmdevtools ncurses-devel pesign grubby libXi-devel gcc-c++ git ccache flex bison elfutils-libelf-devel elfutils-devel openssl openssl-devel dwarves rpm-build perl-devel perl-generators python3-devel make -y ${clang_deps} -y
+    fi
+  elif [ "$_distro" = "Suse" ]; then
+    msg2 "Installing dependencies"
+    sudo zypper install -y rpmdevtools ncurses-devel pesign libXi-devel gcc-c++ git ccache flex bison elfutils libelf-devel openssl-devel dwarves make patch bc rpm-build libqt5-qtbase-common-devel libqt5-qtbase-devel lz4 ${clang_deps}
   fi
+}
+
+_linux_git_branch_checkout() {
+
+  cd "$_where"
+
+  if ! [ -d linux-src-git ]; then
+    msg2 "First initialization of the linux source code git folder"
+    mkdir linux-src-git
+    cd linux-src-git
+    git init
+    git remote add origin https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
+  else
+    cd linux-src-git
+
+    msg2 "Current branch: $(git branch | grep "\*")"
+    msg2 "Reseting files to their original state"
+    
+    git reset --hard HEAD
+    git clean -f -d -x
+  fi 
+
+  if [[ "$_sub" = rc* ]]; then
+    msg2 "Switching to master branch for RC Kernel"
```

Comment:
Afaik master is ahead of RC releases so maybe it could be split up to `rc` and `master`? Maybe using tags for it would be the solution

Reply:
Yes the script is doing is: switch to master, shallow clone the past 3 weeks, then checkout the RC tag in that time period where it's expected to be. Do you have a better idea on how to handle it ?

Reply:
Sorry, overlooked that RC tag checkout

Reply:
No worries!